File: LeetCode/Easy/0746-min-cost-climbing-stairs/0746-min-cost-climbing-stairs.java
class Solution {
    public int minCostClimbingStairs(int[] cost) {
        int n = cost.length;
        int[] dp = new int[n];
        
        // 첫 번째 인덱스 초기화
        dp[0] = cost[0]; 
        // 두 번째 인덱스 초기화
        dp[1] = cost[1]; 
        
        for (int i = 2; i < n; i++) {
            // i번째까지 도달하는 데 필요한 최소 비용
            dp[i] = Math.min(dp[i-1], dp[i-2]) + cost[i];
        }
        
        // 마지막 계단 또는 마지막에서 두 번째 계단 중 최소 비용 리턴
        return Math.min(dp[n-1], dp[n-2]);
    }
}

File: LeetCode/Easy/0746-min-cost-climbing-stairs/0746-min-cost-climbing-stairs.kt
class Solution {
    fun minCostClimbingStairs(cost: IntArray): Int {
        val resultCost = IntArray(cost.size + 1) { 0 }

        for (i in cost.indices) {
            resultCost[i] = cost[i]
        }
        if (cost.size == 1) {
            return cost[0]
        }

        for (i in cost.size - 2 downTo 0) {
            resultCost[i] = (resultCost[i] + resultCost[i + 1]).coerceAtMost(resultCost[i] + resultCost[i + 2])
        }

        return resultCost[0].coerceAtMost(resultCost[1])
    }
}